lib/llm/src/kv_router/indexer.rs

```diff
@@ -1304,65 +1318,107 @@ mod tests {
         let token = CancellationToken::new();
-        let duration = Some(Duration::from_millis(50));
+        let expiration = Duration::from_millis(50);
 
         if num_shards == 1 {
             kv_indexer = Box::new(KvIndexer::new_with_frequency(
                 token,
-                duration,
+                Some(expiration),
                 kv_block_size,
             ));
         } else {
             kv_indexer = Box::new(KvIndexerSharded::new_with_frequency(
                 token,
                 num_shards,
-                duration,
+                Some(expiration),
                 kv_block_size,
             ));
         }
 
-        let worker_id = 0;
-
-        let event = create_store_event(worker_id, 0, vec![1, 2, 3, 4], None);
-        kv_indexer.apply_event(event).await;
-
-        time::sleep(Duration::from_millis(5)).await;
-
+        // The blocks
         let block_hashes = vec![
             LocalBlockHash(1),
             LocalBlockHash(2),
             LocalBlockHash(3),
             LocalBlockHash(4),
         ];
-        let scores = kv_indexer.find_matches(block_hashes.clone()).await.unwrap();
 
-        assert_eq!(scores.frequencies.len(), 0);
+        let overlap = kv_indexer.find_matches(block_hashes.clone()).await.unwrap();
+        assert_eq!(
+            overlap.frequencies.len(),
+            0,
+            "Should be no cached blocks yet"
+        );
+
+        // Blocks go in cache
+        let worker_id = 0;
+        let event = create_store_event(worker_id, 0, vec![1, 2, 3, 4], None);
+        kv_indexer.apply_event(event).await;
 
-        let scores = kv_indexer.find_matches(block_hashes.clone()).await.unwrap();
-        assert_eq!(scores.frequencies, vec![1, 1, 1, 1]);
+        // First access
+        // The store event is applied async so poll briefly
+        let mut overlap = OverlapScores::default();
+        let timeout = Duration::from_millis(10);
+        let start = Instant::now();
+        while overlap.scores.is_empty() && Instant::now().duration_since(start) < timeout {
+            time::sleep(ONE_MILLIS).await;
+            overlap = kv_indexer.find_matches(block_hashes.clone()).await.unwrap();
+        }
+        assert_eq!(
+            overlap.scores.len(),
+            1,
+            "One worker has these blocks cached"
+        );
+        assert_eq!(
+            overlap.frequencies.len(),
+            0,
+            "Blocks have not previously been accessed"
+        );
+
+        // Second access
+        let overlap = kv_indexer.find_matches(block_hashes.clone()).await.unwrap();
+        assert_eq!(overlap.scores.len(), 1, "Still one worker matches");
+        assert_eq!(
+            overlap.frequencies,
+            vec![1, 1, 1, 1],
+            "We should see the first access now"
+        );
 
-        time::sleep(Duration::from_millis(100)).await;
+        // Let those two accesses expire
+        time::sleep(expiration + Duration::from_millis(10)).await;
```

Comment:
Would something like expiration+timeout here be clearer? Assuming the + 10ms here is to match timeout

Reply:
On second read, maybe this just needs to sleep for expiration + some number to increase consistency?  Wasn't immediately clear of the importance of +10ms specifically vs. any other number at a glance

Reply:
The 10ms is any value to put us beyond the expiration. 1ms would have sufficed. It's expiration + "a bit".